<?xml version="1.0" ?>

<launch>
  <!-- Arguments -->
  <arg name="map_file" default="$(find navigation)/maps/my_world_map2.yaml"/>  
  <arg name="move_forward_only" default="true"/>
  <arg name="waypoints_file" default="$(find navigation)/waypoints/waypoints.txt"/>

  <!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>

  <!-- AMCL -->
  <include file="$(find navigation)/launch/amcl.launch"/>

  <!-- move_base -->
  <include file="$(find navigation)/launch/move_base.launch">    
    <arg name="move_forward_only" value="$(arg move_forward_only)"/>
  </include>

  <!-- get_amcl_pose_server (server) -->
  <node name="get_amcl_pose_server" pkg="navigation" type="get_amcl_pose_server_node" />

  <!-- get_pixel_pose_server (server) -->
  <node name="get_pixel_pose_server" pkg="navigation" type="get_pixel_pose_server_node">
    <param name="map_yaml_path" value="$(arg map_file)" />
  </node>

  <!-- pixel to map converter (server) -->
  <node name="convert_pixel_to_map_server" pkg="navigation" type="convert_pixel_to_map_server_node">
    <param name="map_yaml_path" value="$(arg map_file)" />
  </node>

  <!-- pixel waypoints to map waypoints converter (server) -->
  <node name="convert_pixel_waypoints_to_map_waypoints_server" pkg="navigation" type="convert_pixel_waypoints_to_map_waypoints_server_node">
    <param name="map_yaml_path" value="$(arg map_file)" />
  </node>

  <!-- pixel position publisher (Publisher) -->
  <node name="pixel_position_publisher" pkg="navigation" type="pixel_position_publisher_node">
    <param name="map_yaml_path" value="$(arg map_file)" />
  </node>

  <!-- pixel position publisher (Publisher) -->
  <node name="waypoint_plotter_publisher" pkg="navigation" type="waypoint_plotter_publisher_node">
    <param name="waypoints_file" value="$(arg waypoints_file)" />
  </node>

  <!-- robot previous path node -->
  <node name="robot_path_publisher" pkg="bumperbot_utils" type="path_plotter_node" />

  <!-- robot controller node -->
  <node name="robot_controller_node" pkg="navigation" type="robot_controller_node.py" />   

  <!-- Setup waypoint manager -->
  <node name="waypoint_manager_node" pkg="navigation" type="waypoint_manager_node"/>


  <!-- rviz -->
  <node 
    pkg="rviz" 
    type="rviz" 
    name="rviz" 
    args="-d $(find navigation)/rviz/navigation.rviz"
    required="true"    
  />  
</launch>